pylint/extensions/private_import.py

```diff
@@ -185,7 +185,7 @@ def _populate_type_annotations_function(
 
     def _populate_type_annotations_annotation(
         self,
-        node: nodes.Attribute | nodes.Subscript | nodes.Name,
+        node: nodes.Attribute | nodes.Subscript | nodes.Name | None,
```

Comment:
@Pierre-Sassoulas Are you sure this is correct? I think we get `None` because of incorrect typing in `astroid`. Looking at the code it should never be `None`.

Reply:
None is handled correctly inside the function though. if we have None we return None.